### PR TITLE
LibGfx: Draw glyph runs with subpixel accuracy

### PR DESCRIPTION
**Pull request by gmta**
This improves the quality of our font rendering, especially when animations are involved. Relevant changes:

  * Skia fonts have their subpixel flag set, which means that individual glyphs are rendered at subpixel offsets causing glyph runs as a whole to look better.

  * Fragment offsets are no longer rounded to whole device pixels, and instead the floating point offset is kept. This allows us to pass through the floating point baseline position all the way to the Skia calls, which already expected that to be a float position.

The `scrollable-contains-table.html` ref test needed different table headings since they would slightly inflate the column size in the test file, but not the reference.

The improvement is clearly visible when hovering the pictures on https://aaronfrancis.com and looking at the caption below the picture:

| Before | After |
|---|---|
| <video width="320" src="https://github.com/user-attachments/assets/0d12fbf6-803c-482f-b272-ccd5e70a21e0"/> | <video width="320" src="https://github.com/user-attachments/assets/e4d7114d-2481-4048-8ffc-0092874e05a1"/> |

### CODE REVIEW

File: Libraries/LibWeb/Layout/LineBoxFragment.cpp [hunk 1]
```diff
@@ -5,10 +5,8 @@
  */
 
 #include <AK/Utf8View.h>
-#include <LibWeb/DOM/Range.h>
```
Comment (shlyakpavel) [inline Libraries/LibWeb/Layout/LineBoxFragment.cpp:8]: 👍

**Comment by shlyakpavel** [comment]
This looks like a great thing to do!
However, how confident are we that float point operations are consistent enough across different machines to pass the tests reliably? I haven't delved deeply into our test suite, so I'm unsure if it compares exact float point values or uses some form of proximity-based evaluation

**Comment by shlyakpavel** [comment]
Opennet.ru on master looks like this
telegram-cloud-photo-size-2-5384083531417906322-y
The same page with this PR branch looks like this
telegram-cloud-photo-size-2-5384083531417906321-y
Is this introducing some kind of regression?

**Comment by shlyakpavel** [comment]
This PR looks like this on my Mac with this PR applied
image

**Comment by gmta** [comment]
@shlyakpavel 

> However, how confident are we that float point operations are consistent enough

We already do floating point comparisons for things like layout box dimensions by comparing the layout tree output, for example. If we run into issues later, we can deal with them then.

> Is this introducing some kind of regression?

Seems like it - I've not seen anything like that on Linux. I'll build on macOS and report back.

**Comment by gmta** [comment]
@shlyakpavel I messed up text painting when scaling comes into play, which is typically the case on macOS. This has now been fixed.

**Pull request merged**